.github/workflows/build-dev.yaml

```diff
@@ -25,74 +25,71 @@ env:
   GIT_VERSION: 1
 
 jobs:
-  linux-x64:
+  build:
     runs-on: ubuntu-latest
+    container: defi/ain-builder:latest
+    strategy:
+      matrix:
+        target: [x86_64-pc-linux-gnu, x86_64-w64-mingw32, x86_64-apple-darwin]
     env:
-      TARGET: x86_64-pc-linux-gnu
+      TARGET: ${{matrix.target}}
 
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
```

Comment:
Think we can use the options for checkout action to `set-safe-directory: *` instead of the manual command.

Reply:
This will not work. Seems like set-safe-directory is a bool config which defaults to true. Ref: https://github.com/actions/checkout/pull/770/.

Reply:
Seems like it is a known issue that when running jobs inside docker containers, which makes the config not work as desired. The known workaround is to explicitly call git config after the action checkout.